packages/next/src/client/performance-relayer.ts

```diff
@@ -1,102 +0,0 @@
-// TODO: remove in the next major version
-/* global location */
-import type { Metric, ReportCallback } from 'next/dist/compiled/web-vitals'
-
-// copied to prevent pulling in un-necessary utils
-const WEB_VITALS = ['CLS', 'FCP', 'FID', 'INP', 'LCP', 'TTFB']
-
-const initialHref = location.href
-let isRegistered = false
-let userReportHandler: ReportCallback | undefined
-type Attribution = (typeof WEB_VITALS)[number]
-
-function onReport(metric: Metric): void {
-  if (userReportHandler) {
-    userReportHandler(metric)
-  }
-
-  // This code is not shipped, executed, or present in the client-side
-  // JavaScript bundle unless explicitly enabled in your application.
-  //
-  // When this feature is enabled, we'll make it very clear by printing a
-  // message during the build (`next build`).
-  if (
-    process.env.NODE_ENV === 'production' &&
-    // This field is empty unless you explicitly configure it:
-    process.env.__NEXT_ANALYTICS_ID
-  ) {
-    const body: Record<string, string> = {
-      dsn: process.env.__NEXT_ANALYTICS_ID,
-      id: metric.id,
-      page: window.__NEXT_DATA__?.page,
-      href: initialHref,
-      event_name: metric.name,
-      value: metric.value.toString(),
-      speed:
-        'connection' in navigator &&
-        (navigator as any)['connection'] &&
-        'effectiveType' in (navigator as any)['connection']
-          ? ((navigator as any)['connection']['effectiveType'] as string)
-          : '',
-    }
-
-    const blob = new Blob([new URLSearchParams(body).toString()], {
-      // This content type is necessary for `sendBeacon`:
-      type: 'application/x-www-form-urlencoded',
-    })
-    const vitalsUrl = 'https://vitals.vercel-insights.com/v1/vitals'
-    // Navigator has to be bound to ensure it does not error in some browsers
-    // https://xgwang.me/posts/you-may-not-know-beacon/#it-may-throw-error%2C-be-sure-to-catch
-    const send = navigator.sendBeacon && navigator.sendBeacon.bind(navigator)
-
-    function fallbackSend() {
-      fetch(vitalsUrl, {
-        body: blob,
-        method: 'POST',
-        credentials: 'omit',
-        keepalive: true,
-        // console.error is used here as when the fetch fails it does not affect functioning of the app
-      }).catch(console.error)
-    }
-
-    try {
-      // If send is undefined it'll throw as well. This reduces output code size.
-      send!(vitalsUrl, blob) || fallbackSend()
-    } catch (err) {
-      fallbackSend()
-    }
-  }
-}
-
-export default (onPerfEntry?: ReportCallback): void => {
-  // Update function if it changes:
-  userReportHandler = onPerfEntry
```

Comment:
**question** I think if we remove this lines we will also remove the support for https://nextjs.org/docs/app/api-reference/functions/use-report-web-vitals

Reply:
We're safe: the [hook](https://github.com/vercel/next.js/blob/0b3ead5e6bbf5629afc111a400b845576fe7ec20/packages/next/src/client/web-vitals.ts#L12) is completely independent from this file.

We still have passing tests for this hook:
- [hook used](https://github.com/feugy/next.js/blob/damien/aly-1257-remove-speed-insights-from-nextjs/test/e2e/app-dir/app/app/report-web-vitals/reporter.js)
- [test file](https://github.com/feugy/next.js/blob/damien/aly-1257-remove-speed-insights-from-nextjs/test/e2e/app-dir/app/useReportWebVitals.test.ts#L20)